src/versions.cr

```diff
@@ -146,10 +146,19 @@ module Shards
       end
     end
 
+    def self.prerelease?(str)
+      str.each_char.any?(&.ascii_letter?)
```

Comment:
Hyphens are often also used to determine pre-releases. Rubygems considers any version with a letter or hyphen as pre-release.
This obviously only makes a difference when the pre-release tag doesn't contain any letters (like `0.1-1`), which is probably pretty rare.

Reply:
Does it really? Sigh. `0.1.1-1` looks like a build number that supersedes a `0.1.1` release, it doesn't look like a pre-release to me.

Shall we really support that?

Reply:
Rubygems doesn't look for a hyphen, only for a letter:
https://github.com/rubygems/rubygems/blob/master/lib/rubygems/version.rb#L290-L297

Reply:
I'm not sure if this needs to be supported. But rubygems made extra effort to achieve this, so maybe they have a good reason?

Anyway, it's really trivial to implement here, so why not?

Reply:
Sigh. I really start to hate semver altogether. The first draft was okay, but it's really become ugly.

Reply:
> it's really trivial to implement here

Actually, it's not, because dashes and dots are equivalent, so `1.0.0-1` == `1.0.0.1`. I must use the same trick than rubygems and replace `-` with a `.pre.` so `1.0.0-1` becomes `1.0.0.pre.1` to be considered a prerelease, and thus older to `1.0.0`.

Reply:
The replace leads to oddities, like `0.0.1-alpha > 0.0.1.beta` because we parse `0.0.1-alpha` as `0.0.1.pre.alpha` but don't transform `0.0.1.beta`.

We should probably check the partition result (dot or dash) in addition to the segment, but... I honestly, I'm tired, here.

Reply:
If we don't currently support semver perfectly then we don't have to support it perfectly in this PR.

Reply:
btw, `n.n.n-n` notation is used by homebrew, where last part denotes _revision_, far from being pre-release.

Reply:
@Sija This is pretty common with any package distribution. Crystal's apt and rpm packages also use this release + iteration format.
But in semver, everything following a hyphen makes a pre-release. We can't fit both. But I agree with @RX14.

So every version containing a letter is considered a pre-release.